core/unit_test/view/TestBasicView.hpp

```diff
@@ -0,0 +1,249 @@
+  for (int r = 0; r < static_cast<int>(view_type::rank()); r++)
+    expected_size *= view.extent(r);
+  EXPECT_EQ(view.size(), expected_size);
+  EXPECT_EQ(view.empty(), expected_size == 0u);
+}
+
+template <class ViewType>
+struct MDRangeTestFunctor {
+  ViewType view;
+  template <class... Idxs>
+  KOKKOS_FUNCTION void operator()(Idxs... idxs) const {
+    view(idxs...) = (idxs + ...);
+  }
+};
+
+template <class T, class LayoutType, class ExtentsType>
+static void test_access_with_extents(const ExtentsType &extents) {
+  using extents_type  = ExtentsType;
+  using layout_type   = Kokkos::Experimental::layout_right_padded<>;
+  using accessor_type = Kokkos::Impl::CheckedReferenceCountedAccessor<
+      T, typename ExecutionSpace::memory_space>;
+  using view_type =
+      Kokkos::BasicView<T, extents_type, layout_type, accessor_type>;
+
+  auto view = view_type("test_view", extents);
+
+  EXPECT_TRUE(view.data_handle().has_record());
+  EXPECT_NE(view.data_handle().get(), nullptr);
+
+  auto mdrange_policy =
+      make_spanning_mdrange_policy_from_extents<ExecutionSpace>(extents);
+
+  Kokkos::parallel_for(mdrange_policy, MDRangeTestFunctor<view_type>{view});
+}
+
+#if 0  // TODO: this test should be active after View is put on top of BasicView
+  template <class T, template <std::size_t> class LayoutType, class SrcViewType,
+            class ExtentsType>
+  static void test_construct_from_view(const ExtentsType &extents,
+                                       std::size_t padding) {
+    using extents_type  = ExtentsType;
+    using layout_type   = LayoutType<Kokkos::dynamic_extent>;
+    using mapping_type  = typename layout_type::template mapping<ExtentsType>;
+    using accessor_type = Kokkos::Impl::CheckedReferenceCountedAccessor<
+        T, typename ExecutionSpace::memory_space>;
+    using basic_view_type =
+        Kokkos::BasicView<T, extents_type, layout_type, accessor_type>;
+    using view_type = SrcViewType;
+    static_assert(std::is_constructible_v<basic_view_type, SrcViewType>);
+  }
+#endif
+
+template <class T, class LayoutType>
+static void test_access() {
+  test_access_with_extents<T, LayoutType>(Kokkos::extents<std::size_t, 5>());
+}
+
+template <class T>
+static void test_atomic_accessor() {
+  using extents_type = Kokkos::extents<int, 10, 12, 30>;
+  using layout_type  = Kokkos::Experimental::layout_right_padded<>;
+  using accessor_type =
+      Kokkos::Impl::CheckedReferenceCountedRelaxedAtomicAccessor<
+          T, typename ExecutionSpace::memory_space>;
+  using view_type =
+      Kokkos::BasicView<T, extents_type, layout_type, accessor_type>;
+
+  extents_type extents{};
+  auto view = view_type("test_view", extents);
+
+  EXPECT_TRUE(view.data_handle().has_record());
+  EXPECT_NE(view.data_handle().get(), nullptr);
+
+  auto mdrange_policy =
+      make_spanning_mdrange_policy_from_extents<ExecutionSpace>(extents);
+
+  Kokkos::parallel_for(mdrange_policy, MDRangeTestFunctor<view_type>{view});
```

Comment:
What is this test achieving?

Reply:
testing that the access operator compiles with the number of arguments specified.